src/shape.cpp

```diff
@@ -706,9 +706,10 @@ void migraphx_from_value(const value& v, shape& s)
         {
             auto v_dd = v.at("dynamic_dimensions");
             std::vector<shape::dynamic_dimension> dyn_dims(v.at("dynamic_dimensions").size());
-            std::transform(v_dd.begin(), v_dd.end(), dyn_dims.begin(), [](migraphx::value x) {
-                return from_value<shape::dynamic_dimension>(x);
-            });
+            std::transform(
+                v_dd.begin(), v_dd.end(), dyn_dims.begin(), [](const migraphx::value& x) {
+                    return from_value<shape::dynamic_dimension>(x);
+                });
```

Comment:
I think this whole thing can be replaced with `from_value(v.at("dynamic_dimensions"), dyn_dims)` or `auto dyn_dims = from_value<std::vector<shape::dynamic_dimension>>(v.at("dynamic_dimensions"))`.